components/tidb_query/src/batch/runner.rs

```diff
@@ -40,12 +39,6 @@ pub struct BatchExecutorsRunner<SS> {
     // TODO: Deprecate it using a better deadline mechanism.
     deadline: Deadline,
```

Comment:
Can deadline also be extracted?

Reply:
We can. But I'm hesitating integrating to `ConcurrencyLimiter` or adding another interceptor for it.

This PR is fairly large now. I don't intend to add more to it.